File: Programowanie Obiektowe Projekt Nr 3 Python/Sheep.py
#-------------------------------------------------------------------------------
# Name:        module1
# Purpose:
#
# Author:      Kontekstowy
#
# Created:     24-05-2016
# Copyright:   (c) Kontekstowy 2016
# Licence:     <your licence>
#-------------------------------------------------------------------------------
from Animal import Animal
class Sheep(Animal):
    #print(self._x)
    def __init__(self,xx,yy,world,numberx,tempSign):
        Animal.__init__(self,xx,yy,world,numberx,tempSign)
        self._initiative=4
        self._strength=4
        self._sign='S'

    def action(self,animalBoard,nx,ny,animalContainer,board):
        if animalBoard[self._x  + nx][self._y +ny] == -1:
            return 1
        if animalBoard[self._x + nx][self._y + ny] != -1:
            if animalBoard[self._x + nx][self._y + ny] == 'F':
                print("Owca zabija lisa")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == 'T':
                print("Owca atakuje zolwia , ale zostala odparta")
                return 0
            if animalBoard[self._x + nx][self._y + ny] == 'W':
                print("Owca atakuje wilka , ale zostala zabita")
                return 3
            if animalBoard[self._x + nx][self._y + ny] == 'A':
                print("Owca atakuje antylope i ja zabija")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == 'H':
                print("Owca atakuje czlowieka , ale zostala zabita")
                return 3
            if animalBoard[self._x + nx][self._y + ny] == '#':
                print("Owca zjada trawe")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == 'G':
                print("Owca zjada Guarane")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == 'N':
                print("Owca zjada trujace wilcze jagody i ginie")
                return 5
            if animalBoard[self._x + nx][self._y + ny] == 'L':
                print("Owca zjada mleczyk")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == 'S':
                if nx !=0 and ny !=0:
                    return 4
                else:
                    return 0

File: Programowanie Obiektowe Projekt Nr 2 Java/src/objectiveworld160553/Sheep.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package objectiveworld160553;

/**
 *
 * @author Kontekstowy
 */
public class Sheep extends Animal {

    public Sheep(int x, int y, World world) {
        super(x, y, world);
        this.name = "S";
        this.initiative = 4;
        this.strength = 4;
    }

    @Override
    public int action(char animalBoard[][], int nx, int ny, Organism me) {
        switch (animalBoard[x + nx][y + ny]) {
            case '0':
                return 1;
            case 'S':
             //   System.out.println("Spotkaly sie 2 owce");
                return 4;
            case 'T':
                System.out.println("Zolw odpiera owce");
                return 0;
            case 'N':
                System.out.println("Owca zjada wilcze jagody i ginie");
                return 5;
            case 'W':
                System.out.println("Wilk zabija atakujaca owce");
                return 3;
            case 'F':
                System.out.println("Spotyka sie antylopa z owca");
                return 0;
            case 'L':
                System.out.println("Owca zjada mlecz");
                return 2;
            case '#':
                System.out.println("Owca zjada trawe");
                return 2;
            case 'G':
                System.out.println("Owca zjadla guaraney");
                return 2;
            case 'H':
                System.out.println("Owca ginie atakujac czlowieka");
                return 3;
        }

        return 0;
    }
}
